File: 2018/07-jul/7-29-overloading/swift/Main.swift
/*
 Create a garden to demonstrate method overloading in Swift
 - Author: Andrew Jarombek
 - Date: 7/28/2018
 */

let msGroundhog = Animal(name: "Ms. Groundhog", species: AnimalSpecies.Groundhog,
                         description: "Enjoys lounging and eating grass all day")
let mrGroundhog = Animal(name: "Mr. Groundhog", species: AnimalSpecies.Groundhog)

let doe = Animal(name: "doe", species: AnimalSpecies.Deer,
                 description: "from the tip of his wand burst the silver doe")
let bunny = Animal(name: "bunny", species: AnimalSpecies.Rabbit)

let hosta = Plant(species: PlantSpecies.Hosta)
let lily = Plant(species: PlantSpecies.Daylily, in_bloom: true)
let iris = Plant(species: PlantSpecies.Iris, in_bloom: false)

let garden = Garden(withAnimals: [msGroundhog, mrGroundhog, doe, bunny], plants: [hosta, lily, iris], andObjects: nil)

assert(garden.inGarden(doe))

let otherDoe: Any = doe
assert(!garden.inGarden(otherDoe))

assert(garden.inGarden(lily))
assert(!garden.inGarden(Animal(name: "momma rabbit", species: AnimalSpecies.Rabbit)))
assert(!garden.inGarden("Squirrel"))

File: 2018/07-jul/7-29-overloading/php/main.php
<?php
/**
 * Demonstrate overloading in PHP
 * User: Andrew Jarombek
 * Date: 7/28/2018
 */

include 'Animal.php';
include 'AnimalSpecies.php';
include 'Plant.php';
include 'PlantSpecies.php';
include 'Garden.php';

$msGroundhog = new Animal("Ms. Groundhog", AnimalSpecies::GROUNDHOG,
                            "Enjoys lounging and eating grass all day");
$mrGroundhog = new Animal("Mr. Groundhog", AnimalSpecies::GROUNDHOG);

$doe = new Animal("doe", AnimalSpecies::DEER, "from the tip of his wand burst the silver doe");
$bunny = new Animal("bunny", AnimalSpecies::RABBIT);

$hosta = new Plant(PlantSpecies::HOSTA);
$lily = new Plant(PlantSpecies::DAYLILY);
$iris = new Plant(PlantSpecies::IRIS);

$garden = new Garden([$msGroundhog, $mrGroundhog, $doe, $bunny], [$hosta, $lily, $iris]);

assert($garden->inGarden($doe));
assert($garden->inGarden($lily));
assert(!$garden->inGarden(new Animal("momma rabbit", AnimalSpecies::RABBIT)));